ompi/util/timings.h

```diff
@@ -4,123 +4,215 @@
+    while( NULL != t->next ) { t = t->next; };                                 \
+    t->next = (ompi_timing_list_t*)malloc(sizeof(ompi_timing_list_t));         \
+    if ( NULL == t->next) {                                                    \
+        printf("OMPI_TIMING [%d %s]: error memory allocation\n",               \
+            __LINE__, __FUNCTION__);                                           \
+        abort();                                                               \
+    }                                                                          \
+    t->next->next = NULL;                                                      \
+    OMPI_TIMING_ITEM_INIT(&t->next->item);                                     \
+})
+
+#define OMPI_TIMING_ENV_FINALIZE ({                                            \
+    __label__ exit;                                                            \
+    ompi_timing_list_t *t = OMPI_TIMING_env.timing, *tmp;                      \
+    if ( NULL == t) { goto exit; }                                             \
+    do {                                                                       \
+        tmp = t;                                                               \
+        t = t->next;                                                           \
+        if ( tmp->item ) {                                                     \
+            if ( tmp->item->desc ) {                                           \
+                int i;                                                         \
+                for (i = 0; i < tmp->item->use; i++) {                         \
+                    free(tmp->item->desc[i]);                                  \
+                    tmp->item->desc[i] = NULL;                                 \
+                }                                                              \
+                free(tmp->item->desc);                                         \
+                tmp->item->desc = NULL;                                        \
+            }                                                                  \
+            if ( tmp->item->in ) { free(tmp->item->in); }                      \
+            free(tmp->item);                                                   \
+            tmp->item = NULL;                                                  \
+        }                                                                      \
+        free(tmp);                                                             \
+        tmp = NULL;                                                            \
+    } while ( NULL != t);                                                      \
+exit:;                                                                         \
+    OMPI_TIMING_env.cnt = 0;                                                   \
+})
+
+#define OMPI_TIMING_INIT(_inum)                                                \
```

Comment:
I want user to be able to specify the name, not `OMPI_TIMING_env`
Also - OMPI timings are not env-based.
So no need in "env" in the name.